src/OpenIddict.Core/OpenIddictProvider.cs

```diff
@@ -234,6 +240,55 @@ public class OpenIddictProvider<TUser, TApplication> : OpenIdConnectServerProvid
             context.HandleResponse();
         }
 
+        public override async Task ProfileEndpoint([NotNull] ProfileEndpointContext context) {
+            var manager = context.HttpContext.RequestServices.GetRequiredService<OpenIddictManager<TUser, TApplication>>();
+
+            var principal = context.AuthenticationTicket?.Principal;
```

Comment:
nit: what's the benefit of ```?.``` if we assert this should not be null and call ```GetUserId()``` extension method on the instance?